src/sentry/static/sentry/app/views/releasesV2/list/index.tsx

```diff
@@ -131,6 +129,37 @@ class ReleasesList extends AsyncView<Props, State> {
     );
   }
 
+  renderLoading() {
+    return this.renderBody();
+  }
+
+  renderEmptyMessage() {
+    const {location} = this.props;
+    const searchQuery = this.getQuery();
+
+    if (searchQuery && searchQuery.length) {
+      return (
+        <EmptyStateWarning small>{`${t(
+          'There are no releases that match'
+        )}: '${searchQuery}'.`}</EmptyStateWarning>
+      );
+    }
+
+    if (this.getSort() !== 'date') {
+      const relativePeriod = getRelativeSummary(
+        location.query.statsPeriod || DEFAULT_STATS_PERIOD
+      ).toLowerCase();
+
+      return (
+        <EmptyStateWarning small>
+          {`${t('There are no releases with data in the')} ${relativePeriod}.`}
```

Comment:
`relativePeriod` can be `'Other'` or `'Invalid period'`.

Reply:
The `GlobalSelectionHeader` on this page is set to not allow anything other than predefined periods and we don't handle if someone manually writes something stupid in URL (because backend will fail anyway).